config/src/main/java/org/springframework/security/config/annotation/web/configurers/saml2/Saml2LoginConfigurer.java

```diff
@@ -211,11 +226,16 @@ public void init(B http) throws Exception {
 	public void configure(B http) throws Exception {
 		http.addFilter(this.authenticationRequestEndpoint.build(http));
 		super.configure(http);
+		if (this.authenticationManager == null) {
+			saml2WebSsoAuthenticationFilter.setAuthenticationManager(getAuthenticationManager());
+		}
+		else {
+			saml2WebSsoAuthenticationFilter.setAuthenticationManager(this.authenticationManager);
+		}
 	}
 
-	private AuthenticationProvider getAuthenticationProvider() {
-		AuthenticationProvider provider = new OpenSamlAuthenticationProvider();
-		return postProcess(provider);
+	private AuthenticationManager getAuthenticationManager() {
```

Comment:
I think this method should probably register the provider using `http.authenticationProvider()` and then return the shared `AuthenticationManager`. The default behavior for these configurers is to register a provider with the shared provider manager. If the code calls `saml2Login().authenticationManager()`, it's clear that they want to replace this default behavior for this filter.

Reply:
Simplified it to a return void and renamed it to 
[`registerDefaultAuthenticationProvider`](https://github.com/fhanik/spring-security/commit/146502a967b5c0d2c95796ba8a5300c52dac65ff)